etcdmain/config.go

```diff
@@ -152,6 +152,8 @@ func newConfig() *config {
 
 	fs.StringVar(&cfg.Dproxy, "discovery-proxy", cfg.Dproxy, "HTTP proxy to use for traffic to discovery service.")
 	fs.StringVar(&cfg.DNSCluster, "discovery-srv", cfg.DNSCluster, "DNS domain used to bootstrap initial cluster.")
+	fs.StringVar(&cfg.DNSClusterServiceNameSSL, "discovery-srv-name-ssl", cfg.DNSClusterServiceNameSSL, "Service name to query when using DNS discovery for SSL")
```

Comment:
Can we reduce the configurable flag down to one? The goal is to be able to have N etcd discovery under one domain, not to make the ssl suffix configurable?

Reply:
Yes we already know if etcd is using TLS at this point right, how can we get this information from etcd? Can you name your etcd service `_etcd-server-ssl` then in the case of cluster 'a' cumulative SRV would be `_etcd-server-ssl-a._tcp.service.consul. using `--discovery-srv-name a?